plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestCachingDirectoryLister.java

```diff
@@ -0,0 +1,282 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.MaterializedRow;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.hive.HiveQueryRunner.TPCH_SCHEMA;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static java.nio.file.Files.createTempDirectory;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestCachingDirectoryLister
+        extends AbstractTestQueryFramework
+{
+    private CachingDirectoryLister cachingDirectoryLister;
+    private FileHiveMetastore fileHiveMetastore;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Path temporaryMetastoreDirectory = createTempDirectory(null);
+        closeAfterClass(() -> deleteRecursively(temporaryMetastoreDirectory, ALLOW_INSECURE));
+
+        cachingDirectoryLister = new CachingDirectoryLister(Duration.valueOf("5m"), 1_000_000L, List.of("tpch.*"));
+
+        return HiveQueryRunner.builder()
+                .setHiveProperties(ImmutableMap.of(
+                        "hive.allow-register-partition-procedure", "true"))
+                .setMetastore(distributedQueryRunner -> fileHiveMetastore = createTestingFileHiveMetastore(temporaryMetastoreDirectory.toFile()))
+                .setCachingDirectoryListerModule(binder -> {
+                    binder.bind(DirectoryLister.class).toInstance(cachingDirectoryLister);
+                    binder.bind(DirectoryCallback.class).toInstance(cachingDirectoryLister);
+                })
+                .build();
+    }
+
+    @Test
+    public void testCacheInvalidationIsAppliedSpecificallyOnTheTableBeingChanged()
+    {
+        String table1Name = "cached_table1_" + randomTableSuffix();
```

Comment:
table name randomization is unnecessary here, as the test executes on isolated resources

remove randomization and inline the table name in SQL statements, for readability